x-pack/plugins/secops/public/components/timeline/data_providers/data_provider.ts

```diff
@@ -6,10 +6,12 @@
 
 import * as React from 'react';
 
+export type DataProviderId = string;
```

Comment:
You can do this and redefine string, but if you do I will point out all the other areas you missed using this and instead defined id as just a string such as:

```ts
export interface IdToDataProvider {
  [id: string]: DataProvider;
}
```

I would suggest just removing this and using a normal string